### PR TITLE
Move flag util pkgs to k8s.io/component-base/cli

### PR DESCRIPTION
**Pull request by Klaven**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind cleanup

**What this PR does / why we need it**:

This PR is in direct response the the creation of the component-base WG and is work that is described in the backlog. This PR moves the `apiserver/pkg/util/[flag|globalflag]` -> `component-base/cli/[flag|globalflag]`

**Which issue(s) this PR fixes**:

Fixes none# 

**Special notes for your reviewer**:
This PR has 3 commits, The first is code changes, the second is auto gen'ed files and the last is updating lint exceptions

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

**Comment by Klaven** [comment]
hmmmmm...... I ran `./hack/verify-godeps.sh` and it passed on my machine. am I missing something?

File: test/e2e_node/services/kubelet.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ import (
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
-	utilflag "k8s.io/apiserver/pkg/util/flag"
+	utilflag "k8s.io/component-base/cli/flag"
```
Comment (luxas) [inline test/e2e_node/services/kubelet.go]: maybe componentflag or something else?

**Review by luxas — COMMENTED** [review]
Thanks @Klaven!
This looks like a pretty clean move 😄

Two improvements yet please:
 - We should settle on one import name for `k8s.io/component-base/cli/flag`. Whether that's `componentflag`/`cliflag`/`utilflag` I don't really care, but we should be consistent across the board.
 - the `TerminalSize` function in `sanctioned.go` _should not_ be moved (yet at least). Instead, move that func (only) to `k8s.io/apiserver/pkg/util/term`. This is due to that it pulls in a lot of other godeps, and we want to try to add as little deps as just possible for the moment. And it ain't directly related to flags in any case so :)

File: test/e2e_node/services/kubelet.go (hunk 1 above)
Comment (Klaven) [reply test/e2e_node/services/kubelet.go]: componentflag seems long too me, what about `cliflag`?

Comment (Klaven) [reply test/e2e_node/services/kubelet.go]: This will also incur much larger code changes through-out instead of just imports.

**Comment by thockin** [comment]
It looks like you have great reviewers.   Assign to me if you need approval.

File: cmd/cloud-controller-manager/app/controllermanager.go [hunk 2]
```diff
@@ -30,11 +30,11 @@ import (
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/apiserver/pkg/server"
 	"k8s.io/apiserver/pkg/server/healthz"
-	apiserverflag "k8s.io/apiserver/pkg/util/flag"
-	"k8s.io/apiserver/pkg/util/globalflag"
 	"k8s.io/client-go/tools/leaderelection"
 	"k8s.io/client-go/tools/leaderelection/resourcelock"
 	cloudprovider "k8s.io/cloud-provider"
+	apiserverflag "k8s.io/component-base/cli/flag"
+	"k8s.io/component-base/cli/globalflag"
```
Comment (stewart-yu) [inline cmd/cloud-controller-manager/app/controllermanager.go]: can we rename 'apiserverflag'?
the same in others place

File: cmd/cloud-controller-manager/app/options/options.go [hunk 3]
```diff
@@ -22,18 +22,18 @@ import (
 	"net"
 	"time"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (stewart-yu) [inline cmd/cloud-controller-manager/app/options/options.go]: keep original better, and others place

File: cmd/kube-apiserver/app/options/options_test.go [hunk 4]
```diff
@@ -27,11 +27,11 @@ import (
 	"k8s.io/apimachinery/pkg/util/diff"
 	apiserveroptions "k8s.io/apiserver/pkg/server/options"
 	"k8s.io/apiserver/pkg/storage/storagebackend"
-	utilflag "k8s.io/apiserver/pkg/util/flag"
 	auditbuffered "k8s.io/apiserver/plugin/pkg/audit/buffered"
 	auditdynamic "k8s.io/apiserver/plugin/pkg/audit/dynamic"
 	audittruncate "k8s.io/apiserver/plugin/pkg/audit/truncate"
 	restclient "k8s.io/client-go/rest"
+	utilflag "k8s.io/component-base/cli/flag"
 	"k8s.io/kubernetes/pkg/api/legacyscheme"
```
Comment (stewart-yu) [inline cmd/kube-apiserver/app/options/options_test.go]: rename `utilflag`, ans others place

File: cmd/kube-apiserver/app/server.go [hunk 5]
```diff
@@ -35,7 +35,7 @@ import (
 	"github.com/spf13/cobra"
 
 	extensionsapiserver "k8s.io/apiextensions-apiserver/pkg/apiserver"
-	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
```
Comment (stewart-yu) [inline cmd/kube-apiserver/app/server.go]: keep original

File: test/e2e_node/services/kubelet.go (hunk 1 above)
Comment (stewart-yu) [reply test/e2e_node/services/kubelet.go]: IMO, using `componentflag` and `componentglobalflag` in all change

File: cmd/kube-apiserver/app/server.go (hunk 5 above)
Comment (luxas) [reply cmd/kube-apiserver/app/server.go]: this is autogenerated and does no harm

File: cmd/cloud-controller-manager/app/options/options.go (hunk 3 above)
Comment (Klaven) [reply cmd/cloud-controller-manager/app/options/options.go]: I'm not sure how this happened. nice catch.

File: test/e2e_node/services/kubelet.go (hunk 1 above)
Comment (Klaven) [reply test/e2e_node/services/kubelet.go]: after talking with luxas I changed them all to cliflag.

File: cmd/kube-apiserver/app/server.go (hunk 5 above)
Comment (Klaven) [reply cmd/kube-apiserver/app/server.go]: I think this is the last thing. please let me know if I need to go back and update this.

File: cmd/cloud-controller-manager/app/controllermanager.go (hunk 2 above)
Comment (Klaven) [reply cmd/cloud-controller-manager/app/controllermanager.go]: done. it is now `cliflag`. On a side note. I did not name any of the `globalflag` imports as it was never named before and left those imports bare.

**Comment by Klaven** [comment]
will look into bazel-build failure in the morning. I ran `./hack/update-bazel.sh` must have missed running something else.

**Comment by Klaven** [comment]
@luxas I have corrected the issues and got all the failing tests working except the `ete-containerized` one and I think that one is ok?

**Comment by Klaven** [comment]
I will squash the PR when you all are finished looking over it.

File: staging/src/k8s.io/kube-aggregator/Godeps/Godeps.json [hunk 6]
```diff
@@ -6,14 +6,6 @@
 		"./..."
 	],
 	"Deps": [
-		{
```
Comment (luxas) [inline staging/src/k8s.io/kube-aggregator/Godeps/Godeps.json]: Great to cut this unnecessary dependency!

File: hack/.golint_failures [hunk 7]
```diff
@@ -644,6 +643,7 @@ staging/src/k8s.io/code-generator/cmd/client-gen/types
 staging/src/k8s.io/code-generator/cmd/conversion-gen/generators
 staging/src/k8s.io/code-generator/cmd/go-to-protobuf/protobuf
 staging/src/k8s.io/code-generator/cmd/lister-gen/generators
+staging/src/k8s.io/component-base/cli/flag
```
Comment (luxas) [inline hack/.golint_failures]: This is ok, but eventually we should fix this

**Review by luxas — APPROVED** [review]
/lgtm
/approve

Thanks @Klaven! This looks like a clean move of packages :+1:

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 8]
```diff
@@ -34,8 +34,8 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
 	"k8s.io/apimachinery/pkg/util/sets"
-	"k8s.io/apimachinery/pkg/util/uuid"
 	"k8s.io/apimachinery/pkg/util/term"
+	"k8s.io/apimachinery/pkg/util/uuid"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/apiserver/pkg/server"
```
Comment (stewart-yu) [inline cmd/kube-controller-manager/app/controllermanager.go]: i wonder why this change?

File: staging/src/k8s.io/apiserver/pkg/util/term/term.go [hunk 9]
```diff
@@ -36,4 +36,4 @@ func TerminalSize(w io.Writer) (int, int, error) {
 		return 0, 0, err
 	}
 	return int(winsize.Width), int(winsize.Height), nil
-}
```
Comment (stewart-yu) [inline staging/src/k8s.io/apiserver/pkg/util/term/term.go]: why this change in this commit?

File: cmd/cloud-controller-manager/app/controllermanager.go [hunk 10]
```diff
@@ -26,8 +26,8 @@ import (
 	"github.com/spf13/cobra"
 
 	"k8s.io/apimachinery/pkg/util/sets"
-	"k8s.io/apimachinery/pkg/util/uuid"
 	"k8s.io/apimachinery/pkg/util/term"
```
Comment (stewart-yu) [inline cmd/cloud-controller-manager/app/controllermanager.go]: the same

**Review by stewart-yu — COMMENTED** [review]
i think we should make the commit log more graceful:sweat_smile:

**Comment by rosti** [comment]
/assign @thockin 
for approval, as suggested by @luxas on Slack.

**Review by thockin — COMMENTED** [review]
/lgtm
/approve

**Comment by rosti** [comment]
@Klaven the PR needs rebasing

**Comment by nikhita** [comment]
Cancelling lgtm to stop retesting, as this PR needs a rebase.

/lgtm cancel
